src/PlatformCrypto.c

```diff
@@ -68,7 +70,8 @@ bool PltEncryptMessage(PPLT_CRYPTO_CONTEXT ctx, int algorithm, int flags,
     outLength = *outputDataLength;
 
     if (tag != NULL) {
-        if (mbedtls_cipher_auth_encrypt(&ctx->ctx, iv, ivLength, NULL, 0, inputData, inputDataLength, outputData, &outLength, tag, tagLength) != 0) {
+        if (mbedtls_gcm_crypt_and_tag(ctx->ctx.cipher_ctx, MBEDTLS_GCM_ENCRYPT, inputDataLength, iv, ivLength, NULL, 0,
```

Comment:
It looks like these functions are going away soon too, so we may just need to go back to your original approach :(

https://github.com/Mbed-TLS/mbedtls/issues/1674

Reply:
I see. So If I understand correctly, in GFE's payload the tag is BEFORE data so multiple extra `memmove`s are non-avoidable :(

Reply:
Yeah, it looks like all uses of GCM have the tag before the ciphertext. At least they're consistent 😆